controllers/secretagentconfiguration_controller.go

```diff
@@ -179,20 +187,26 @@ func (reconciler *SecretAgentConfigurationReconciler) Reconcile(req ctrl.Request
 			keyRefSecrets := make(map[string][]byte)
 			refs, refDataKeys := keyInterface.References()
 			for index, ref := range refs {
-
-				secRefObject, err := k8ssecrets.LoadSecret(reconciler.Client, ref, instance.Namespace)
-
-				if err != nil {
-					log.Error(err, "error looking up secret ref",
-						"secret_name", secretReq.Name,
-						"secret_ref", ref)
-					rescheduleRetry, errorFound = true, true
+				var secRefObject *corev1.Secret
+				// if ref the current secret, use the values already loaded in memory, else load them from k8s
+				if ref == secObject.Name {
```

Comment:
I think most of the block around LOC 205 can be in this if statement, and use the break here. 

This should let you remove the else LOC 194 and make it a little more concise.